### ISSUE TITLE
Error loading provider when using 'phpipam_address' resource

### ISSUE THREAD

**Issue by brianjgrier** [Ouest-France/terraform-provider-phpipam#34]
Attempting to us V0.6.0 but receive an error during terraform init if my Template includes 'resource "phpipam_address" "my_vm_addr"'

brian:setup_poc_vm$ terraform version
Terraform v0.13.5
+ provider registry.terraform.io/hashicorp/vsphere v1.24.2
+ provider registry.terraform.io/ouest-france/phpipam v0.6.0

Your version of Terraform is out of date! The latest version
is 0.14.2. You can update by downloading from https://www.terraform.io/downloads.html
brian:setup_poc_vm$

### When there is no resource included in the template:

 brian:setup_poc_vm$ terraform init
Initializing modules...
- create_systems in modules/create_vm_group

Initializing the backend...

Successfully configured the backend "local"! Terraform will automatically
use this backend unless the backend configuration changes.

Initializing provider plugins...
- Finding ouest-france/phpipam versions matching "0.6.0"...
- Finding hashicorp/vsphere versions matching "~> 1.24.2"...
- Installing ouest-france/phpipam v0.6.0...
- Installed ouest-france/phpipam v0.6.0 (self-signed, key ID A2243135B89F0CD5)
- Installing hashicorp/vsphere v1.24.2...
- Installed hashicorp/vsphere v1.24.2 (signed by HashiCorp)

Partner and community providers are signed by their developers.
If you'd like to know more about provider signing, you can read about it here:
https://www.terraform.io/docs/plugins/signing.html

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.

### When I uncomment the following section in the template:

// Reserve the address
   resource "phpipam_address" "vmx_addrs" {
     subnet_id   = data.phpipam_subnet.subnet.subnet_id
     hostname    = "my_vm"
     description = "Managed by Terraform"
}

### and re-run iterraform init:

brian:setup_poc_vm$ terraform init
Initializing modules...

Initializing the backend...

Initializing provider plugins...
- Using previously-installed ouest-france/phpipam v0.6.0
- Using previously-installed hashicorp/vsphere v1.24.2
- Finding latest version of hashicorp/phpipam...

Error: Failed to install provider

Error while installing hashicorp/phpipam: provider registry
registry.terraform.io does not have a provider named
registry.terraform.io/hashicorp/phpipam

brian:setup_poc_vm$

I removed the exiting .terraform directory and re-ran init

brian:setup_poc_vm$ rm -rf .terraform/
brian:setup_poc_vm$ terraform init
Initializing modules...
- create_systems in modules/create_vm_group

Initializing the backend...

Successfully configured the backend "local"! Terraform will automatically
use this backend unless the backend configuration changes.

Initializing provider plugins...
- Finding ouest-france/phpipam versions matching "0.6.0"...
- Finding hashicorp/vsphere versions matching "~> 1.24.2"...
- Finding latest version of hashicorp/phpipam...
- Installing ouest-france/phpipam v0.6.0...
- Installed ouest-france/phpipam v0.6.0 (self-signed, key ID A2243135B89F0CD5)
- Installing hashicorp/vsphere v1.24.2...
- Installed hashicorp/vsphere v1.24.2 (signed by HashiCorp)

Partner and community providers are signed by their developers.
If you'd like to know more about provider signing, you can read about it here:
https://www.terraform.io/docs/plugins/signing.html

Error: Failed to install provider

Error while installing hashicorp/phpipam: provider registry
registry.terraform.io does not have a provider named
registry.terraform.io/hashicorp/phpipam

brian:setup_poc_vm$

**Comment by pablo-ruth** [comment]
Terraform is searching for the wrong path:
```
Finding latest version of hashicorp/phpipam...
```
If you are migrating from Terraform 0.12 you have to do a "replace-provider" command to fix path in state:
```
terraform013 state replace-provider "hashicorp/phpipam" "registry.terraform.io/ouest-france/phpipam"
```
Do you have a versions.tf file? Can you paste your provider config and the output of the "terraform providers" command?

**Comment by brianjgrier** [comment]
Late last night I found my mistake.

I am using modules in my template. I had defined the provider information in the root module only.

This is a simplified view:

```
.
├── get_ipaddr.tf
└── modules
    └── allocate_ip
        └── allocat.tf
```


My mistake was that in the called module I did not have "required_providers" block. Adding that to the called module fixed the problem.

It appears it needs to be in every called module that uses your provider.

Sorry to have bothered you.
